docs/DeveloperGuide.md

```diff
@@ -1,38 +1,275 @@
+endpoint is reached at which point the corresponding `ExecutableCommandFactory` is returned.
+
+![RouterSnippetUML.png](images/RouterSnippetUML.png)
+
+This is an object diagram representing a portion of the routing tree for MealCompanion, supporting the "recipe all",
+"recipe possible", and "ingredient list" commands.
+
+The sequence diagram below illustrates the process for resolving the "recipe all" command.
+
+![RouterSequenceUML.png](images/RouterSequenceUML.png)
+
+### Ingredient Class
+
+Below shows the class diagram of how ingredients are being stored in our program
+
+![IngredientUML.png](images/IngredientUML.png)
+
+The current `MealCompanionSession` would keep track of the `IngredientList` which is an ArrayList of `Ingredient` objects. Each `Ingredient` object has a name and its associated quantity. This association would allow us to perform the adding and removal of ingredients by the user.
+
+### Recipe Class
+
+Below shows the class diagram of how recipes are being stored in our program
+
+![RecipeUML.png](images%2FRecipeUML.png)
+
+The current `MealCompanionSession` would keep track of the `RecipeList` which is an ArrayList of `Recipe` objects.
+
+## Implementation
+
+### Add and Remove Command
+
+The add and remove command is facilitated by the methods in `IngredientList` and `Ingredient`. Given below is an example usage scenario and how the add and remove command behaves.
+
+Step 1. The user inputs his command e.g. `add egg /qty 5`, `IngredientList` would be called to check if egg is already stored inside the list
+
+Step 2. Suppose egg is not currently stored in `IngredientList`,  the name of the ingredient 'egg' would be crosschecked with our database of known ingredients
+
+Step 3. Since egg is part of our list of known ingredients, a new `Ingredient` object would be created with the quantity, 5, and name, egg, specified by the user and added to `IngredientList` with the `add()` method.
+
+Below shows the sequence diagram for the above AddCommand:
+
+![AddIngredientSequenceUML.png](images/AddIngredientSequenceUML.png)
+
+Continuing from the above AddCommand:
+
+Step 4. Now the user decides to remove 2 eggs and inputs his command e.g. `remove egg /qty 2`, the `IngredientList` would be searched through to see if egg is in the list
+
+Step 5. Since egg is in the list, its corresponding index in the list would be generated. 
+
+Step 6. The current quantity of egg in the list would be obtained by calling `getQuantity()` and checked to see if it is greater or equals to '2' the quantity input by the user
+
+Step 7. Since the quantity input by the user is smaller than the current quantity of egg, which is 3, in the `IngredientList`, the new quantity would be calculated to be 3 and updated by calling `setQuantity(3)` 
+
+Below shows the sequence diagram for the above RemoveCommand:
+
+![RemoveIngredientSequenceUML.png](images/RemoveIngredientSequenceUML.png)
+
+###### [Back to table of contents](#table-of-contents)
```

Comment:
I like the usage of "Back to the table of contents" so that readers can find their way back to the "Table of contents" with ease!